### ISSUE TITLE
New report for tax purposes

### ISSUE THREAD

**Issue by lukeskinner** [lukeskinner/OTHub#63]

**Comment by lukeskinner** [comment]
telegram with removed names:


For only completed jobs?


I think the problem atm is, you cant JUST export completed job if you didnt payed them out



Also if you were able to pull the usd value at the time of completion. It's really going to be more useful when there are more jobs


Hopefully the automatic payout setting is chain-by-chain and not a universal setting for the node. I would like XDAI and SFC jobs to payout automatically and ETH jobs to not. I hope this is possible.


Call it your tax export and i would pay for that




that's what the View Payouts in USD report was supposed to be. But I guess you want it so that it can:
A. include jobs not paid out yet
B. have the ability to price the payout in USD at the time of job completion

If that's right I'll log it on github for now?


obvs I won't break the existing way, just give more options


We want to export only completed jobs. Even if they were paid out yet


If I can pull that info from an othub api in could probs build a script that'll pull it down into csv. Never done something like that so you might get it done quicker/better.


Basically what we need to tax report is  
the Day of the finished job, 
the amount of trac and 
the ID. ( US Value is a bonus ontop)


And this for all completed jobs from one node ( dosnt metter if payedout or not)


i think i'm following, payout isn't important for tax purposes for you. i'll log this on github, about 75% of api calls on othub support exporting to csv/json automatically so it's not too much extra effort to supply this


this would probs be a new page similar to view payouts in usd

**Comment by lukeskinner** [comment]
Notes of proposed page below...

### User can pick to calculate USD price based on
- (1 row per offer) Price at job start
- (1 row per offer) Price at job completion
- (1 row per payout) Price at job payouts from blockchain

### Filters the user can change

- Checkbox for including active jobs
- Checkbox for including completed jobs

- Filter for start/end date on offers
- Filter for a specific node or all nodes

### Nice to have features

- Ability to save settings
- Ability to save reports

### Columns to show in data
- Date for start or completion or payout
- Offer ID
- Amount in TRAC
- Amount in USD

### Total numbers to show from data
- Total amount in TRAC
- Total amount in USD

**Comment by lukeskinner** [comment]
Early proof of concept available at https://v5.othub.info/nodes/mynodes/taxreport

**Comment by lukeskinner** [comment]
feedback:
`Would it be worth having a job duration column? Some countries may tax jobs longer than a year different to short jobs. `

**Comment by lukeskinner** [comment]
1. The screen was cut off and there was no scroller on my computer, managed to scroll to the right by fiddling and dragging my mouse around. 2. There is no End Date, this is more important than Start Date. 3. Need a total or the ability to export to CSV (I think you said above this is on the list). 4. Would it be worth having a job duration column? Some countries may tax jobs longer than a year different to short jobs. 5. Should there be the ability to select which nodes it includes? Some people may run the OThub page for a friend mixed in with theirs? Or some might have a business managing nodes for people? (Again you said above this is coming) 6. Add a way to filter columns maybe? Or the CSV would solve it. Overall I think it is amazing

**Comment by lukeskinner** [comment]
unfinished bits / on the todo list (happy to add more):
1. Test the hell out of the different modes / confirm the data that comes back
2. Show totals for TRAC and USD so you don't have to add it all up manually
3. Add a node filter instead of it being stuck doing all your nodes
4. Make it prettier + fix formatting of some data
